backend/api/views.py

```diff
@@ -1,3 +1,96 @@
-from django.shortcuts import render
+from rest_framework.views import APIView
+import api.models as myModels
+import api.serializers as mySerializers
+from rest_framework import generics, permissions
+from rest_framework.response import Response
+from rest_framework.authtoken.models import Token
+from .serializers import UserSerializer, LoginSerializer
+import datetime
+
 
 # Create your views here.
+class CoursePlanList(APIView):
```

Comment:
mozna tu po prostu wrzucic ListApiView z generics lub po prostu `viewsets.ModelViewSet`, wterdy nie musi bawic sie z pisaniem get'a.

Reply:
Jesli chce ograniczyc zeby tylko byl get to `mixins.ListModelMixin` i `GenericViewSet` powinno starczyc

Reply:
np cos takiego:
```
from rest_framework.permissions import IsAuthenticated
from rest_framework import mixins, viewsets

class CoursePlanList(mixins.RetrieveModelMixin, mixins.ListModelMixin, GenericViewSet):
    queryset = CoursePlan.objects.all()
    serializer_class = CoursePlanSerializer
    permission_classes = [IsAuthenticated]
```

Reply:
@mima-design  You can use triple "ˋˋˋ" with word python (ˋˋˋpythonˋˋˋ) to make it better-looking:
```python
from rest_framework.permissions import IsAuthenticated
from rest_framework import mixins, viewsets

class CoursePlanList(mixins.RetrieveModelMixin, mixins.ListModelMixin, GenericViewSet):
    queryset = CoursePlan.objects.all()
    serializer_class = CoursePlanSerializer
    permission_classes = [IsAuthenticated]
```

Reply:
jak próbuję tak zrobić całe api się wysypuje

Reply:
@Wojtop Przydałoby się to rozwiązać, tutaj @mima-design ma całkowitą rację tutaj - lepiej użyć dedykowanego sposobu. Jakiś błąd konkretny?
Spróbuj zamienić `ApiView` na `ListApiView`:

```python
from rest_framework.generics import ListAPIView
class CoursePlanList(ListAPIView):
    queryset = CoursePlan.objects.all()
    serializer_class = CoursePlanSerializer
    permission_classes = [IsAuthenticated]
```
powinno zadziałać, a jak nie to opisz co się tam dzieje.

Reply:
Tak, ta wersja zadziałała, dzięki (zostawię bez permission classes, bo te rzeczy mogą być dostępne też bez logowania i będą potrzebne z tego co wiem).